### PR TITLE
Ft client signup

### PR DESCRIPTION
**Pull request by AbonyiXavier**
## Description
Client sign-up endpoint was added and validation using joi

All pull requests should be related to open issues. Indicate the issue(s) bellow and remove this line:
Fixes #(issue number 1) #(issue number 2 if applicable)

## How Has This Been Tested?
Please describe in detail how you tested your changes.
Include details of your testing environment, and the tests you ran to see how your change affects other areas of the code etc.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added necessary inline code documentation
- [ ] I have added tests that prove my fix is effective and that this feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -7,8 +7,8 @@ import cors from 'cors';
 import helmet from 'helmet';
 // import db from './models';
 
-import indexRouter from './routes/index';
-import usersRouter from './routes/users';
+// import indexRouter from './routes/index';
+// import usersRouter from './routes/users';
```
Comment (Kibetchirchir) [inline server/app.js]: Remove code if you dont need it rather than commenting

File: server/controllers/userSignUp.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import express from 'express';
+import bcrypt from 'bcryptjs';
+import jwt from 'jsonwebtoken';
+const models = require('../models');
+const signupValidation = require('../validation/userSignup');
```
Comment (Kibetchirchir) [inline server/controllers/userSignUp.js]: be constant use es6 in your file

File: server/controllers/userSignUp.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    });
+    await models.Clients.findOne({
+      where: {
+        email: req.body.email,
+      },
+    });
+    if (!clientData) {
+      res.send({
+        message: 'client not found',
+      });
```
Comment (Kibetchirchir) [inline server/controllers/userSignUp.js]: why are you checking on the client again

File: server/controllers/userSignUp.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    const clientData = await models.Clients.create({
+      firstName: req.body.firstName,
+      lastName: req.body.lastName,
+      email: req.body.email,
+      password: bcrypt.hashSync(req.body.password, 10),
+      phoneNumber: req.body.phoneNumber,
+      address: req.body.address,
+      isAdmin: req.body.isAdmin,
+    });
+    await models.Clients.findOne({
```
Comment (Kibetchirchir) [inline server/controllers/userSignUp.js]: model should be in  singular form

File: server/validation/userSignup.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import Joi from 'joi';
+
+const signupValidation = (data) => {
+  const schema = {
+    firstName: Joi.string().min(4).max(150).required(),
+    lastName: Joi.string().min(4).max(150).required(),
+    email: Joi.string().email().min(3).max(150),
```
Comment (Kibetchirchir) [inline server/validation/userSignup.js:7]: nice implementation

**Comment by Kibetchirchir** [comment]
LGTM just fix the codacy error

File: server/controllers/userSignUp.js (hunk 3 above)
Comment (AbonyiXavier) [reply server/controllers/userSignUp.js]: I will remove it then

File: server/controllers/userSignUp.js (hunk 2 above)
Comment (AbonyiXavier) [reply server/controllers/userSignUp.js]: Done on this

File: server/controllers/userSignUp.js (hunk 4 above)
Comment (AbonyiXavier) [reply server/controllers/userSignUp.js]: Done on this as well

File: server/validation/userSignup.js (hunk 5 above)
Comment (AbonyiXavier) [reply server/validation/userSignup.js:7]: Thank you sir

**Review by sadiqasg — DISMISSED** [review]
well done Xavier

**Pull request merged**